src/main/java/core/basesyntax/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+import core.basesyntax.exception.EntityNotFoundException;
+import core.basesyntax.mapper.ShoppingCartMapper;
+import core.basesyntax.mapper.UserMapper;
+import core.basesyntax.model.CartItem;
+import core.basesyntax.model.ShoppingCart;
+import core.basesyntax.model.User;
+import core.basesyntax.repository.cartitem.CartItemRepository;
+import core.basesyntax.repository.shoppingcart.ShoppingCartRepository;
+import core.basesyntax.service.ShoppingCartService;
+import jakarta.transaction.Transactional;
+import java.util.HashSet;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository repository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final UserMapper userMapper;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    public void delete(Long cartId) {
+        repository.deleteById(cartId);
+    }
+
+    @Override
+    public ShoppingCartDto updateCartItemQuantity(User user, UpdateCartItemQuantityDto request) {
+        Long bookId = request.getBookId();
+        int quantity = request.getQuantity();
+
+        ShoppingCart cart = repository.findByUser(user)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "ShoppingCart not found for user: " + user.getUsername()));
+
+        Optional<CartItem> cartItem = cartItemRepository.findByShoppingCartAndBook_Id(cart, bookId);
+
+        if (cartItem.isPresent()) {
+            cartItem.get().setQuantity(quantity);
+        } else {
+            throw new EntityNotFoundException("CartItem with bookId "
+                    + bookId + " not found in cart");
+        }
+
+        ShoppingCart updatedCart = repository.save(cart);
+        return shoppingCartMapper.toDto(updatedCart);
+    }
+
+    @Override
+    public ShoppingCartDto updateCartItemQuantity(User user, Long cartItemId, int quantity) {
+        ShoppingCart cart = repository.findByUser(user)
+                .orElseThrow(() ->
+                        new EntityNotFoundException("Cart not found for user: "
+                                + user.getUsername()));
+
+        Optional<CartItem> cartItem = cartItemRepository
+                .findByShoppingCartAndBook_Id(cart, cartItemId);
+        if (cartItem.isPresent()) {
+            cartItem.get().setQuantity(quantity);
+        } else {
+            throw new EntityNotFoundException("CartItem with bookId "
+                    + cartItemId + " not found in cart");
+        }
+        ShoppingCart updated = repository.save(cart);
+        return shoppingCartMapper.toDto(updated);
+    }
+
+    @Override
+    public Page<ShoppingCartDto> getCart(Long cartId, Pageable pageable) {
```

Comment:
Are you getting one cart or getCarts? 
```suggestion
    public ShoppingCartDto getCart(Long cartId, Pageable pageable) {
```

Reply:
As now userId and cartItemId are the same, I will use only Long userId, UpdateCartItemDto itemDto